server/test/handlers/cancelSpecificAppointment.test.js

```diff
@@ -0,0 +1,113 @@
+const workerTimeslot = require('../../models/db/workerTimeslot');
+const appointmentHandler = require('../../models/handlers/appointment');
+const workerTimeslotHandler = require('../../models/handlers/workerTimeslot');
+
+describe('testing cancellation of specific worker appointments', () => {
+    beforeEach(() => {
+        jest.resetModules();
+    });
+
+    test('no parameters passed in for the method', async () => {
+        // Arrange
+        const testIsCancelled = false;
+
+        // Act
+        const isCancelledSuccessfully = await appointmentHandler.cancelSpecificAppointment();
+
+        // Assert
+        expect(isCancelledSuccessfully).toBe(testIsCancelled);
+    });
+
+    test('null values for appointmentId parameter', async () => {
+        // Arrange
+        const appointmentId = null;
+        const testIsCancelled = false;
+
+        // Act
+        const isCancelledSuccessfully = await appointmentHandler.cancelSpecificAppointment(appointmentId);
+
+        // Assert
+        expect(isCancelledSuccessfully).toBe(testIsCancelled);
+    });
+
+    test('empty string for the appointmentId', async () => {
+        // Arrange
+        const appointmentId = '';
+
+        const testIsCancelled = false;
+
+        // Act
+        const isCancelledSuccessfully = await appointmentHandler.cancelSpecificAppointment(appointmentId);
+
+        // Assert
+        expect(isCancelledSuccessfully).toBe(testIsCancelled);
+    });
+
+    test('invalid true value for the appointmentId', async () => {
+        // Arrange
+        const appointmentId = true;
+
+        const testIsCancelled = false;
+
+        // Act
+        const isCancelledSuccessfully = await appointmentHandler.cancelSpecificAppointment(appointmentId);
+
+        // Assert
+        expect(isCancelledSuccessfully).toBe(testIsCancelled);
+    });
+
+    test('invalid number value for the appointmentId', async () => {
+        // Arrange
+        const appointmentId = 37;
+
+        const testIsCancelled = false;
+
+        // Act
+        const isCancelledSuccessfully = await appointmentHandler.cancelSpecificAppointment(appointmentId);
+
+        // Assert
+        expect(isCancelledSuccessfully).toBe(testIsCancelled);
+    });
+
+    test('valid parameters for appointmentId', async () => {
+        // Arrange
+        const appointmentId = 38;
+
+        const testIsCancelled = true;
+
+        // Act
+        const isCancelledSuccessfully = await appointmentHandler.cancelSpecificAppointment(appointmentId);
+
+        // Assert
+        expect(isCancelledSuccessfully).toBe(testIsCancelled);
+    });
+    //Due to database structure, there will never be a worker timeslot id input into this method that is non-existent and therefore testing cases are complete at this point
```

Comment:
Hi Peter,
From this comment, I inferred that the following tests are specifically for a function in the workerTimeslotHandler? If that is a correct assumption, I would suggest moving these tests to a separate file under handlers OR start a new describe section to clearly identify the scope change.

Reply:
describe code has been added to make scope change clear!